### PR TITLE
multilabel notebook

### PR DESCRIPTION
**Pull request by sanderland**
Mostly just required a little dataset map step to format the labels, and some finnicky sampling to ensure the rarer labels are in the subsampled training data.

Considered multi-column labels to the column mapping, but it's more trouble than it's worth in my opinion.

### CODE REVIEW

File: notebooks/text-classification_multilabel.ipynb [hunk 1]
```diff
@@ -0,0 +1,548 @@
+{
```
Comment (lewtun) [inline notebooks/text-classification_multilabel.ipynb:1]: <p>Could we uncomment this line?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/huggingface/setfit/pull/95/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='33'/>

**Review by lewtun — COMMENTED** [review]
Thanks a lot for this clean notebook @sanderland 🔥 !!

I've left a comment about swapping out the dataset - let me know what you think :)

File: notebooks/text-classification_multilabel.ipynb (hunk 1 above)
[the remark this answers is not in the document]
Comment (sanderland) [reply notebooks/text-classification_multilabel.ipynb:1]: Will have a look :)

[the remark this answers is not in the document]
Comment (lewtun) [reply notebooks/text-classification_multilabel.ipynb:1]: Awesome!

[the remark this answers is not in the document]
Comment (sanderland) [reply notebooks/text-classification_multilabel.ipynb:1]: `load_dataset("lex_glue", "ecthr_a")` has lists of texts in the 'text' field. Concatenating these leads to very high GPU ram usage which is not very suitable for an example notebook I think.

[the remark this answers is not in the document]
Comment (lewtun) [reply notebooks/text-classification_multilabel.ipynb:1]: Oh yeah, that subset isn't ideal. What about the `eurlex` [one](https://huggingface.co/datasets/lex_glue/viewer/eurlex/train)? That is a single text and should presumably be truncated if the text is too long. 

Another alternative would be the `ethos` dataset (https://huggingface.co/datasets/ethos/viewer/multilabel/train), which has short texts. The only downside here is that you'll need to create your own validation set, but that could be done with the `train_test_split()` function in `datasets`

Comment (Mouhanedg56) [inline notebooks/text-classification_multilabel.ipynb:1]: <p>It'd be great to add hyperparameter search for <code>num_iterations</code></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/huggingface/setfit/pull/95/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='27'/>

[the remark this answers is not in the document]
Comment (sanderland) [reply notebooks/text-classification_multilabel.ipynb:1]: eurlex has 100 labels, and the training on 8 examples/class takes 2 hours on a 3090.

ethos works nicely!

[the remark this answers is not in the document]
Comment (lewtun) [reply notebooks/text-classification_multilabel.ipynb:1]: Thanks, this looks great!

Comment (lewtun) [reply notebooks/text-classification_multilabel.ipynb:1]: Yeah I agree this is out of scope (but a nice idea nonetheless!). Once #8 is implemented, we could revisit this :)

**Review by lewtun — APPROVED** [review]
Thanks for iterating on the dataset @sanderland - look great 🔥 !
